### DISCUSSION TITLE
I cannot under how does hasura creates migration table behind the scenes

### DISCUSSION THREAD

**Discussion by DVGY** [hasura/graphql-engine#10548] [Question]
Where can I found the relevant code, which explain how hasura maintains the migrations and table schema changes for postgress sql?

**Comment by codedmart** [comment]
@DVGY can you be more specific what it is you are looking for or need an answer on. Here are a few resources that may or may not help:
- https://github.com/hasura/graphql-engine/blob/master/server/documentation/deep-dives/migration-guidelines.md
- https://github.com/hasura/graphql-engine/blob/master/server/src-lib/Hasura/Server/Migrate.hs#L3-L21
